src/Sulu/Bundle/PreviewBundle/Resources/js/containers/Preview/Preview.js

```diff
@@ -85,19 +81,51 @@ class Preview extends React.Component<Props> {
             value: webspace.key,
         }));
 
-        this.previewStore = new PreviewStore(
-            formStore.resourceKey,
-            formStore.id,
-            formStore.locale,
-            this.webspaceKey,
-            this.segments.find((segment) => segment.default === true)?.key
-        );
+        this.createPreviewStore();
 
         if (Preview.mode === 'auto') {
             this.startPreview();
         }
     }
 
+    componentDidUpdate(prevProps: Props) {
+        const {
+            formStore,
+        } = this.props;
+
+        if (this.props.formStore !== prevProps.formStore) {
+            this.dispose();
```

Comment:
would always call the `disposeFormStoreReactions` in the `initializeFormStoreReactions` method